src/native/util/os_linux.cpp

```diff
@@ -61,7 +66,9 @@ ThreadScope::restore_user()
         // must restore uid first otherwise will fail on permission
         MUST_SYS(syscall(SYS_setresuid, -1, orig_uid, -1));
         MUST_SYS(syscall(SYS_setresgid, -1, orig_gid, -1));
-        MUST_SYS(syscall(SYS_setgroups, ThreadScope::orig_groups.size(), &ThreadScope::orig_groups[0]));
+        if(_groups == NULL || orig_groups != *_groups) {
```

Comment:
why do we need this condition? we always want to restore the original groups

Reply:
we can. I just though we wouldn't need to if its already original groups